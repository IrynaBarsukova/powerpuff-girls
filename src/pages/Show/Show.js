import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';

import Container from '../../components/common/Container/Container';
import EntityDetails from '../../components/EntityDetails/EntityDetails';
import EpisodesList from '../../components/EpisodesList/EpisodesList';
import { loadShowStarted } from '../../store/actions/show';
import styles from './Show.scss';

function Show() {
    const { showId } = useParams();
    const dispatch = useDispatch();
    const show = useSelector(state => state.show.data);
    const episodes = useSelector(state => state.show.data.episodes);
    const isLoaded = useSelector(state => state.show.loaded);


    useEffect(() => {
        dispatch(loadShowStarted(showId));
    }, [showId]);

    return (
        <div className={styles.show}>
            {
                isLoaded && (
                    <Container>
                        <EntityDetails entity={show} />
                        <EpisodesList episodes={episodes} />
                    </Container>
                )
            }
        </div>
    );
}

export default Show;
