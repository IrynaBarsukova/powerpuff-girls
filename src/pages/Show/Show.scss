.show {
  padding: 20px 0;
}

.show__title {
  font-weight: normal;
  margin-bottom: 10px;
}

.show__details {
  display: flex;
  justify-content: space-between;
}

.show__summary {
  margin-left: 20px;
}
