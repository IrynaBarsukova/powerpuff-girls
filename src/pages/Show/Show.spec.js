import React from 'react';
import { Provider } from 'react-redux';

import MockStore from '../../mocks/store.mock';
import renderWithRouter from '../../mocks/router.mock';
import mockShow from '../../mocks/show.mock';
import Show from './Show';

describe('<Show />', () => {
    let store;

    beforeEach(() => {
        store = new MockStore({ show: { data: mockShow, loaded: true } });
    });

    it('should render Show', () => {
        const { container } = renderWithRouter(
            <Provider store={store}>
                <Show />
            </Provider>,
            { route: `shows/6771` },
        );

        expect(container).toMatchSnapshot();
    });
});
