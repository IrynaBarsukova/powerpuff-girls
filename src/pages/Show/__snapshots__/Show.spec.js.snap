// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Show /> should render Show 1`] = `
<div>
  <div
    class="show"
  >
    <div
      class="container"
    >
      <div
        class="entity"
      >
        <h2
          class="entity__title"
        >
          The Powerpuff Girls
        </h2>
        <section
          class="entity__details"
        >
          <img
            alt="The Powerpuff Girls"
            class="entity__image"
            src="http://static.tvmaze.com/uploads/images/medium_portrait/60/151357.jpg"
          />
          <p
            class="entity__summary"
          >
            <p>
              The city of Townsville may be a beautiful, bustling metropolis, but don't be fooled! There's evil afoot! And only three things can keep the bad guys at bay: Blossom, Bubbles and Buttercup, three super-powered little girls, known to their fans (and villains everywhere) as 
              <b>
                The Powerpuff Girls
              </b>
              . Juggling school, bedtimes, and beating up giant monsters may be daunting, but together the Powerpuff Girls are up to the task. Battling a who's who of evil, they show what it really means to 'fight like a girl.'
            </p>
          </p>
        </section>
      </div>
      <div
        class="episodes"
      >
        <h3
          class="episodes__title"
        >
          Episodes
        </h3>
        <div
          class="episodesList"
        >
          <div
            class="episodesList__episode"
          >
            <a
              href="shows/shows/6771/episodes/657308"
            >
              <img
                alt="Escape from Monster Island"
                src="http://static.tvmaze.com/uploads/images/medium_landscape/53/132617.jpg"
              />
              <p>
                Escape from Monster Island
              </p>
              <p>
                1
                x
                1
              </p>
            </a>
          </div>
          <div
            class="episodesList__episode"
          >
            <a
              href="shows/shows/6771/episodes/691594"
            >
              <img
                alt="Bye Bye, Bellum"
                src="http://static.tvmaze.com/uploads/images/medium_landscape/53/132631.jpg"
              />
              <p>
                Bye Bye, Bellum
              </p>
              <p>
                1
                x
                7
              </p>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
